packages/ihe-gateway-sdk/src/client/ihe-gateway.ts

```diff
@@ -0,0 +1,44 @@
+import { Patient } from "@medplum/fhirtypes";
+import { v4 as uuidv4 } from "uuid";
+import { principalCareProviderIdsSchema } from "../models/shared";
+import { XCPDGateways, XCPDRequest, xcpdGatewaysSchema } from "../models/xcpd-request";
+
+export class IHEGateway {
+  /**
+   * An XCPD request interface
+   * @param patient The patient data in FHIR R4 format.
+   * @param cxId The customer ID.
+   * @param xcpdGateways The OIDs and XCPD ITI-55 urls of each organization to make a request to.
+   * @param principalCareProviderIds The list of NPIs of the practitioners associated with the patient.
+   * @param requestId Optional. Unique id for the request. If not provided, one will be created.
+   * @returns an XCPD request to be used with an IHE Gateway.
+   */
+  createXCPDRequest({
+    patient,
+    cxId,
+    xcpdGateways,
+    principalCareProviderIds,
+    requestId,
+  }: {
+    patient: Patient;
+    cxId: string;
+    xcpdGateways: XCPDGateways;
+    principalCareProviderIds?: string[];
```

Comment:
shouldn't this be an array of an object created from `principalCareProviderIdsSchema` ?

Reply:
that object is already an array, but ya